Add docker gen2 (#10305)
circleci/circleci-docs: docs/guides/modules/execution-managed/pages/using-docker.adoc

= Using the Docker execution environment
:page-platform: Cloud, Server v4+
:page-description: Learn how to configure a your jobs to run in the Docker execution environment
:page-description: Learn how to configure your jobs to run in the Docker execution environment
:experimental:

> CAUTION: *Legacy images with the prefix `circleci/`` were link:https://discuss.circleci.com/t/legacy-convenience-image-deprecation/41034[deprecated]* on December 31, 2021. For faster builds, upgrade your projects with link:https://circleci.com/blog/announcing-our-next-generation-convenience-images-smaller-faster-more-deterministic/[next-generation convenience images].

Reviewer: Just checking you removed this intentionally? I know its old but I do not know if we still have customers using the legacy images

Author: Yes, this was removed intentionally for that reason.

      - image: cimg/node:lts
----

A container is an instance of a specified Docker image. The first image listed in your configuration for a job is referred to as the _primary_ container image and this is where all steps in the job run. _Secondary_ containers can also be specified to run alongside the primary container for services such as databases.

> NOTE: For an introduction to Docker concepts, see the link:https://docs.docker.com/engine/docker-overview/[Docker Overview documentation].

Reviewer: ```suggestion
TIP: For an introduction to Docker concepts, see the link:https://docs.docker.com/engine/docker-overview/[Docker Overview documentation].
```

include::ROOT:partial$execution-resources/resource-class-view.adoc[]

[#docker-gen2-benefits]
== Docker gen2

NOTE: Docker gen2 resource classes are available on paid plans only. See the xref:plans-pricing:plan-overview.adoc[Plans Overview] for details.

Resource classes running on gen2 infrastructure offer significant performance and cost improvements over their gen1 equivalents.

[#docker-gen2-performance-improvements]
=== Performance improvements

Using more modern compute, gen2 instances are 3x faster than their gen1 equivalents.

> x86 gen2 will be priced higher than its gen1 equivalent. See our link:https://circleci.com/pricing/price-list/[Pricing Page] for details.

Reviewer: ```suggestion
x86 gen2 resources have a higher cost in credits per minute than their gen1 equivalents. See our link:https://circleci.com/pricing/price-list/[Pricing Page] for details.
```

> NOTE: Gen2 does not currently support the xref:security:ip-ranges.adoc[IP Ranges].

Reviewer: ```suggestion
NOTE: Gen2 does not currently support the use of xref:security:ip-ranges.adoc[IP Ranges].
```
